Fixes common e2e errors in e2e_learner_flow_skill_editor_and_embedding

kevintab95 opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. ~This PR fixes or fixes part of #[fill_in_number_here].~
2. This PR does the following: This PR fixes the following e2e errors:
- Failed: No element found using locator: By(css selector, .protractor-test-exploration-language-select) [[build](https://github.com/oppia/oppia/pull/11257/checks?check_run_id=1448461047)]
- Failed: unexpected alert open: {Alert text : } [[build](https://github.com/oppia/oppia/pull/11246/checks?check_run_id=1443526699)]


## Essential Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.

Commit:
Fixes the following errors:
1. Failed: No element found using locator: By(css selector, .prot
   ractor-test-exploration-language-select)
2. Failed: unexpected alert open: {Alert text : }

Commit:
fix lint error

U8NWXD reviewed (approved):
Looks good. Thanks for fixing these!

Commit:
add scrollToTop

Commit:
move . before successCallback is called; Few waitFors in settings tab page

Commit:
add more test conditions to ensure changes are published before moving away from the editor page

kevintab95 commented:
Reruns:
1: https://github.com/oppia/oppia/pull/11259/checks?check_run_id=1455830175
![image](https://user-images.githubusercontent.com/11008603/100286641-26a85580-2f99-11eb-9677-0080e97dad0d.png)

2: https://github.com/oppia/oppia/pull/11259/checks?check_run_id=1455987635
![image](https://user-images.githubusercontent.com/11008603/100288778-70933a80-2f9d-11eb-92ec-0b5ec6419700.png)

3: https://github.com/oppia/oppia/pull/11259/checks?check_run_id=1456103613
Failed with "No element found using locator: By(css selector, .protractor-test-select-rubric-difficulty)". Should be fixed by 929b6b8. 

4: https://github.com/oppia/oppia/pull/11259/checks?check_run_id=1456220239
![image](https://user-images.githubusercontent.com/11008603/100292592-ef40a580-2fa6-11eb-9742-720f7fd11546.png)

5: https://github.com/oppia/oppia/pull/11259/checks?check_run_id=1456316133
![image](https://user-images.githubusercontent.com/11008603/100383371-76dbf200-3043-11eb-9f05-e11c7dfd7185.png)

Commit:
Fixes following flake:
  - Failed: No element found using locator: By(css selector, .protractor-test-select-rubric-difficulty)
